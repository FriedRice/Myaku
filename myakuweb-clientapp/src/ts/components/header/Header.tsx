/** @module MyakuWeb main header component */

import React from 'react';

interface HeaderProps {
    children: React.ReactNode;
}

const Header: React.FC<HeaderProps> = function(props) {
    return (
        <header className='page-header'>
            <div className='content-container header-content-container'>
                {props.children}
            </div>
        </header>
    );
};

export default Header;
