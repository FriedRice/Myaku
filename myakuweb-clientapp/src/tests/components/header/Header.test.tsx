/**
 * Tests for the Header component.
 * @module tests/components/header/Header.test
 */

import Header from 'ts/components/header/Header';
import React from 'react';
import { mount } from 'enzyme';


describe('<Header />', function() {
    it('renders correctly', function() {
        const wrapper = mount(
            <Header>
                <div>Header content 1</div>
                <div>Header content 2</div>
            </Header>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
