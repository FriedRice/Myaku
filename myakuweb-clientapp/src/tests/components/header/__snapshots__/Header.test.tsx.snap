// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<Header>
  <header
    className="page-header"
  >
    <div
      className="content-container header-content-container"
    >
      <div>
        Header content 1
      </div>
      <div>
        Header content 2
      </div>
    </div>
  </header>
</Header>
`;
